### PR TITLE
#016/UI/class competitor button

### PR DESCRIPTION
**Pull request by Smyku6**
closes issuse number #16  

image

at this moment it looks like that:
image

I create a two Components strictly connected.

First is the `<ClassCompetitorButton/>`, we can use it to show some data in the class list or in the competitor list.
At this moment I create some test-data so this button read these props from test-data:
1. **type** - string ('class' or 'competitor')
2. **classNumber** - number (number of our class, ex. 'Klasa 0', Klasa 2')
3. **classCompleted** - boolean (if true we can highlight and show text 'completed/ukończono')
4. **contestantsAmount** - number (how many contestants class has),
5. **no** - number (it's an index for every element of the array - in our example it's a list of dogs),
6. **nameOfCompetitor** - string (name of our competitor),
7. **exercisesCompleted** - number (how many exercises is completed by a competitor),
8. **exercisesAmount** - number (how many exercises we have),

This Component renders a div with main text on the left side and inserts another component `<InButtonLabel/>` on the right side.
It passes 5 props to this children Component:
1. **type**,
2. **classCompleted**,
3. **contestantsAmount**,
4. **exercisesAmount**,
5. **exercisesCompleted**,

This `<InButtonLabel/>`  passes some information about completion of class OR completion of exercises to children styled component `<LabelWrapper/>`, so we can highlight when something is completed.

I merge new eslint/prettier config to my branch so all style and syntax problems are fixed now (I must add PropTypes validation).

### CODE REVIEW

**Comment by Smyku6** [comment]
And yes I'm sorry. 
I by accident step into [territory](https://github.com/CC2021-WBL/GOOD-BOI-Application/issues/12)  of our UI-NINJA @pierwszazlewej 
image
She knows about it and I hope she will forgive me. I promise it won't happen again.
image

**Comment by pierwszazlewej** [comment]
No worries, To be honest this time - I'm kinda grateful for this scope trespassing.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (vieraboschkova) [inline .eslintrc.json:1]: This file doesn't belong in this PR

File: .prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (vieraboschkova) [inline .prettierrc.json:1]: This file doesn't belong in this PR

File: package.json [hunk 3]
```diff
@@ -44,6 +44,10 @@
     ]
   },
   "devDependencies": {
-    "babel-plugin-styled-components": "^2.0.2"
+    "babel-plugin-styled-components": "^2.0.2",
```
Comment (vieraboschkova) [inline package.json:47]: These packages don't belong in this PR

**Review by vieraboschkova — DISMISSED** [review]
Please remove files that should not be a part of the scope of this PR

File: src/components/UI/Button/ClassCompetitorButton.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  border-radius: 0.75rem; ;
+`;
+
+const ClassCompetitorButton = (props) => {
+  const {
+    type,
+    classNumber,
+    classCompleted,
+    contestantsAmount,
+    no,
```
Comment (vieraboschkova) [inline src/components/UI/Button/ClassCompetitorButton.js:32]: This might not be necessary. I would imagine this could be a class object coming in that would have the class number like so: `{classNumber: 0, competitors: [array with length you can check], isCompleted: bool, }` or in case of the competitor: `{competitorId: uuid, numerStartowy: givenNumber, name: johnny, exercises: [array you can use later for storing score maybe, ex32: {isComplete: bool, score: 30}]}`

File: src/components/UI/Button/ClassCompetitorButton.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  font-style: normal;
+  line-height: 1.5rem;
+  background: white;
+  border: solid 1px #323f4b;
+  border-radius: 0.75rem; ;
+`;
+
+const ClassCompetitorButton = (props) => {
+  const {
+    type,
```
Comment (vieraboschkova) [inline src/components/UI/Button/ClassCompetitorButton.js:28]: The component should be dynamicly adjusting to the data, so in theory it shouldnt need to get all the information, just the ones that it needs for rendering and then decide what to render depending on what it got

File: src/components/UI/Button/ClassCompetitorButton.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+          {no}. {nameOfCompetitor}
+        </>
+      )}
+
+      <InButtonLabel
+        type={type}
+        classCompleted={classCompleted}
+        contestantsAmount={contestantsAmount}
+        exercisesCompleted={exercisesCompleted}
+        exercisesAmount={exercisesAmount}
```
Comment (vieraboschkova) [inline src/components/UI/Button/ClassCompetitorButton.js:52]: See if you can pass something more generic and then decide what to render within the component

File: src/components/UI/Button/ClassCompetitorButton.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      />
+    </ClassCompetitorButtonWrapper>
+  );
+};
+
+ClassCompetitorButton.propTypes = {
```
Comment (vieraboschkova) [inline src/components/UI/Button/ClassCompetitorButton.js:58]: Great idea using the PropTypes already! Perhaps you could also decide which ones are required

File: src/components/UI/Button/ClassCompetitorButton.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  const {
+    type,
+    classNumber,
+    classCompleted,
+    contestantsAmount,
+    no,
+    nameOfCompetitor,
+    exercisesCompleted,
+    exercisesAmount,
+  } = props;
```
Comment (vieraboschkova) [inline src/components/UI/Button/ClassCompetitorButton.js:36]: Good job! Always destructure props! 👍

File: src/components/UI/Button/InButtonLabel.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+  const exercisesComplete =
+    exercisesAmount !== undefined && exercisesCompleted === exercisesAmount;
+
+  return (
+    <LabelWrapper
+      classCompleted={classCompleted}
+      exerciseCompleted={exercisesComplete}
+    >
+      {type === 'class' && classCompleted && <>ukończono</>}
```
Comment (vieraboschkova) [inline src/components/UI/Button/InButtonLabel.js:42]: Perhaps you can think of a way to remove some written text to consts if you find its either repeating or annoying to write or prone to changes.

**Review by MonikaKrella — APPROVED** [review]
everything is repaired! good work
